ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java

```diff
@@ -759,6 +771,20 @@ public void setSafeBlock(final Hash blockHash) {
     updater.commit();
   }
 
+  private long getFinalizedBlockNumber() {
+    return this.getFinalized()
+        .flatMap(this::getBlockHeader)
+        .flatMap(blockHeader -> Optional.of(blockHeader.getNumber()))
```

Comment:
just saw this little optimization now, to avoid an allocation of `Optional`

```suggestion
        .map(BlockHeader::getNumber)
```